scripts/backup/pg_restore.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+
+###########################
+####### LOAD CONFIG #######
+###########################
+while [ $# -gt 0 ]; do
+    case $1 in
+        -c)
+            if [ -r "$2" ]; then
+                source "$2"
+                shift 2
+            else
+                echo "[ERROR]["$(date +\%Y-\%m-\%d\ %H:%M:%S:%3N)"] Unreadable config file \"$2\"" 1>&2
+                exit 1
+            fi
+            ;;
+        -d)
+            if [ -d "$2" ]; then
+                RESTORE_DIR="$2"
+                shift 2
+            else
+                echo "[ERROR]["$(date +\%Y-\%m-\%d\ %H:%M:%S:%3N)"] Unreadable directory \"$2\"" 1>&2
+                exit 1
+            fi
+            ;;
+        *)
+            echo " [ERROR]["$(date +\%Y-\%m-\%d\ %H:%M:%S:%3N)"] Unknown Option \"$1\"" 1>&2
+            exit 2
+            ;;
+    esac
+done
+echo "FFF $RESTORE_DIR"
+
+if [ $# = 0 ]; then
+    SCRIPTPATH=$(cd ${0%/*} && pwd -P)
+    source $SCRIPTPATH/pg_backup.config
+fi;
+
+###########################
+#### PRE-RESTORE CHECKS ####
+###########################
+
+# Make sure we're running as the required backup user
+if [ "$BACKUP_USER" != "" -a "$(id -un)" != "$BACKUP_USER" ]; then
+	echo "[ERROR]["$(date +\%Y-\%m-\%d\ %H:%M:%S:%3N)"] This script must be run as $BACKUP_USER. Exiting." 1>&2
+	exit 1;
+fi;
+
+if [ -z $RESTORE_DIR ]; then
+	echo "[ERROR]["$(date +\%Y-\%m-\%d\ %H:%M:%S:%3N)"] Backup not provided, please use -d option." 1>&2
+	exit 1;
+fi
+
+if [ $(find $RESTORE_DIR -type f | wc -l) -ne 1 ]; then
+	echo "[ERROR]["$(date +\%Y-\%m-\%d\ %H:%M:%S:%3N)"] Backup directory contains more than one file." 1>&2
+	exit 1;
+fi;
+
+
+###########################
+### INITIALISE DEFAULTS ###
+###########################
+
+if [ ! $HOSTNAME ]; then
+	HOSTNAME="localhost"
+fi;
+
+if [ ! $USR ]; then
+	USR="postgres"
+fi;
+
+
+###########################
+###### FULL BACKUP #######
+###########################
+
+RESTORE_FILE=$(find $RESTORE_DIR -type f)
+
+echo -e "\n\nPerforming restore from $RESTORE_FILE"
+echo -e "--------------------------------------------\n"
+
+echo "Restore $DATABASE"
+if ! zcat $RESTORE_FILE | psql -h "$HOSTNAME" -U "$USR" "$DATABASE"; then
```

Comment:
Are we assuming that access to database allowed by IP address and no password required?

Reply:
@vryazanov It can ask for password on command line (just like pg_dump in pg_backup.sh scripts) - it really depends on the configuration of postres installation. E.g. in my case is asks for password in both cases.

Reply:
@krzyk Yes, I understand, but origin task description says that it's needed for doing backups db by cron. ok, no more questions from me, @emilianodellacasa will decide, mb it's good

Reply:
@vryazanov This depends on the postgres config, you can configure it that way that given user has passwordless access to the database. E.g. you could have a `backup` user in unix, have that cron setup onm that user and postgres configured to allow access for that user.

Reply:
@krzyk would be better to specify user and password

Reply:
@emilianodellacasa I don't think this is a good idea (passwords in files).

If a user want's to do that he can:
1. specify PGPASSWORD environment variable (this one is deprecated)
2. use `~/.pgpass` file
3. Create appropriate config in pg_hba.conf to allow passwordless login

I don't think it is a good idea to add another place for password, considering that such file should have only read permissions for given user. I would stay with pgpass or if needed specify PGPASSWORD in crontab.

Reply:
@krzyk Agree, but please add an explanation on Readme file